library/aria.c

```diff
@@ -112,8 +112,12 @@ static inline uint32_t aria_p1( uint32_t x )
 #endif /* arm */
 #if defined(__GNUC__) && \
     defined(__i386__) || defined(__amd64__) || defined( __x86_64__)
-/* I couldn't find an Intel equivalent of rev16, so two instructions */
-#define ARIA_P1(x) ARIA_P2( ARIA_P3( x ) )
+static inline uint32_t aria_p1( uint32_t x )
+{
+    uint32_t r;
+    __asm( "xchgb %b0, %h0" : "=q"(r) : "0"(x) );
+    return( r );
+}
```

Comment:
You need to also `#define ARIA_P1 aria_p1` here, or the default version below is going to be used. (Clang helpful gives a warning about `aria_p1` being unused, which GCC doesn't.)

Reply:
Note: after adding this line (and enabling ARIA in `config.h`), trying to build the library give the following error with GCC:
```
aria.c: In function ‘aria_fo_xor’:
aria.c:388:1: error: extended registers have no high halves
```
Clang also complains:
```
aria.c:118:12: error: invalid operand for instruction                                                                                                                                                         
    __asm( "xchgb %b0, %h0" : "=q"(r) : "0"(x) );                                                                                                                                                             
           ^                                                                                                                                                                                                  
<inline asm>:1:14: note: instantiated into assembly here                                                                                                                                                      
        xchgb %bpl, %bp                                                                                                                                                                                       
                    ^~~ 
```